tests/manage/mcg/test_bucket_policy.py

```diff
@@ -311,8 +313,414 @@ def test_anonymous_read_only(self, mcg_obj, bucket_factory):
+        assert helpers.s3_get_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name), "Failed: GetBucketWebsite"
+
+        logger.info("Writing index and error data to the bucket")
+        assert helpers.s3_put_object(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+            object_key="index.html", data=index, content_type='text/html'
+        ), "Failed: PutObject"
+        assert helpers.s3_put_object(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+            object_key="error.html", data=error, content_type='text/html'
+        ), "Failed: PutObject"
+
+        # Verifying whether DeleteBucketWebsite action is denied access
+        logger.info(f'Verifying whether user: {obc_obj.obc_account} is denied to DeleteBucketWebsite')
+        try:
+            helpers.s3_delete_bucket_website(s3_obj=obc_obj, bucketname=obc_obj.bucket_name)
+        except boto3exception.ClientError as e:
+            logger.info(e.response)
+            response = HttpResponseParser(e.response)
+            if response.error['Code'] == 'AccessDenied':
+                logger.info('GetObject action has been denied access')
+            else:
+                raise UnexpectedBehaviour(f"Response received invalid error code {response.error['Code']}")
+
+        # Admin modifies policy to allow DeleteBucketWebsite action
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=['DeleteBucketWebsite'],
+            resources_list=[obc_obj.bucket_name, f'{obc_obj.bucket_name}/{"*"}'],
+            effect="Allow"
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {obc_obj.bucket_name} with principal: {obc_obj.obc_account}')
+        assert helpers.put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy), "Failed: PutBucketPolicy"
+
+        # Getting Policy
+        logger.info(f'Getting bucket policy for bucket: {obc_obj.bucket_name}')
+        get_policy = helpers.get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info('Got bucket policy:%s' % get_policy['Policy'])
+
+        logger.info(f"Deleting bucket website config from bucket: {obc_obj.bucket_name}")
+        assert helpers.s3_delete_bucket_website(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name
+        ), "Failed: DeleteBucketWebsite"
+
+    @pytest.mark.polarion_id("OCS-2161")
+    @tier2
+    def test_bucket_versioning_and_policies(self, mcg_obj, bucket_factory):
+        """
+        Tests bucket and object versioning on Noobaa buckets and also its related actions
+        """
+        data = "Sample string content to write to a new S3 object"
+        object_key = "ObjKey-" + str(uuid.uuid4().hex)
+        object_versions = []
+
+        # Creating a OBC user (Account)
+        obc = bucket_factory(amount=1, interface='OC')
```

Comment:
Regarding the interface='OC',
Do we plan to parametrize other interfaces?

Reply:
Other than 'oc' there are 2 more, ie "s3" which creates plain S3 buckets(not account) and "cli" which uses noobaa cli and creates OBC similar to "oc". Since the objective here is to only have an OBC account I chose to go with "oc" interface param.